### PR TITLE
[14.0][REF] shopfloor_reception

### PR DESCRIPTION
**Pull request by vnikolayev1**
- updated shopfloor_reception - changed done_action and move the call of _action_done only into _handle_backorder. This ensures if _handle_backorder is called the picking is set to done

### CODE REVIEW

**Comment by vnikolayev1** [comment]
@mt-software-de

File: shopfloor_reception/services/reception.py [hunk 1]
```diff
@@ -1004,13 +999,16 @@ def done_action(self, picking_id, confirmation=False):
             message=self.msg_store.transfer_done_success(picking)
         )
 
-    def _handle_backorder(self, picking):
+    def _handle_backorder(self, picking, cancel_backorder=False):
```
Comment (mt-software-de) [inline shopfloor_reception/services/reception.py:1002]: You forgot to Set Argument `cancel_backorder` in Line 997

**Review by mt-software-de — CHANGES_REQUESTED** [review]
Thx. There is just a small mistake.

**Review by jbaudoux — COMMENTED** [review]
I don't understand what you are fixing. Can you clarify what was not working?

**Comment by mt-software-de** [comment]
> I don't understand what you are fixing. Can you clarify what was not working?

It is not a fix it is a refactoring. 😂 @vnikolayev1 could please update the title of this PR.

File: shopfloor_reception/services/reception.py [hunk 2]
```diff
@@ -999,18 +994,21 @@ def done_action(self, picking_id, confirmation=False):
             return self._response_for_confirm_done(
                 picking, message=self.msg_store.need_confirmation()
             )
-        self._handle_backorder(picking)
+        self._handle_backorder(picking, True)
```
Comment (mt-software-de) [inline shopfloor_reception/services/reception.py]: We not want to cancel the Backorder Always please Pass the cancel_backorder var. 
```suggestion
        self._handle_backorder(picking, cancel_backorder)
```

**Review by mt-software-de — CHANGES_REQUESTED** [review]

**Comment by jbaudoux** [comment]
> It is not a fix it is a refactoring. 😂 @vnikolayev1 could please update the title of this PR.

A refactor to do what if it is not fixing nor improving anything?

**Comment by mt-software-de** [comment]
> > It is not a fix it is a refactoring. 😂 @vnikolayev1 could please update the title of this PR.
> 
> A refactor to do what if it is not fixing nor improving anything?

It is improving the Code, with this change there is only one call of `picking._action_done`

**Review by mt-software-de — APPROVED** [review]
@jbaudoux could we merge this? Otherwise close it. From our side it is not needed anymore.

**Review by jbaudoux — APPROVED** [review]

**Pull request merged**